### PR TITLE
fix(core): log missing errorInfo in React 18 onRecoverableError callback

### PR DESCRIPTION
**Pull request by johnnyreilly**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

See https://github.com/facebook/docusaurus/issues/9379

## Test Plan

I hope to use this to diagose https://github.com/facebook/docusaurus/issues/9379

## Related issues/PRs

See https://github.com/facebook/docusaurus/issues/9379

### CODE REVIEW

File: packages/docusaurus/src/client/clientEntry.tsx [hunk 1]
```diff
@@ -37,8 +37,8 @@ if (ExecutionEnvironment.canUseDOM) {
     </HelmetProvider>
   );
 
-  const onRecoverableError = (error: unknown): void => {
-    console.error('Docusaurus React Root onRecoverableError:', error);
+  const onRecoverableError = (error: unknown, errorInfo: { digest?: string | null, componentStack?: string | null }): void => {
```
Comment (slorber) [inline packages/docusaurus/src/client/clientEntry.tsx:40]: Thanks, missed that we could get some additional info in the callback

Why not using the official type ReactDOM.ErrorInfo?

**Review by slorber — CHANGES_REQUESTED** [review]
looks like lint/prettier fails

File: packages/docusaurus/src/client/clientEntry.tsx (hunk 1 above)
Comment (johnnyreilly) [reply packages/docusaurus/src/client/clientEntry.tsx:40]: > the official type ReactDOM.ErrorInfo

Oh. where is that type found?  I used the implementation I found in the React codebase - do you have another link?

Comment (johnnyreilly) [reply packages/docusaurus/src/client/clientEntry.tsx:40]: Doesn't show up in the repo? https://github.com/search?q=repo%3Afacebook%2Freact%20ReactDOM.ErrorInfo&type=code

Comment (slorber) [reply packages/docusaurus/src/client/clientEntry.tsx:40]: hmm, will try to fix it

Comment (slorber) [reply packages/docusaurus/src/client/clientEntry.tsx]: probably only exists on external typedefs, fixed

Comment (johnnyreilly) [reply packages/docusaurus/src/client/clientEntry.tsx]: Oh so the type you're using is from DefinitelyTyped: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/29f3f65685aba9e9b8a105a55494923980e2f16e/types/react/index.d.ts#L3240

Ironically, I'm one of the maintainers of that.  It's slightly inaccurate too :-). But it doesn't really matter - we just want the `console.log` to take place

Comment (johnnyreilly) [reply packages/docusaurus/src/client/clientEntry.tsx]: I've raised a PR to update the Definitely Typed definition to be more accurate: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66987

Comment (slorber) [reply packages/docusaurus/src/client/clientEntry.tsx]: yes I know that's why I didn't understand 😅 great if you catched a typo

**Comment by johnnyreilly** [comment]
See https://github.com/facebook/react/issues/26224

**Pull request merged**